base/iobuffer.jl

```diff
@@ -489,6 +489,62 @@ function take!(io::IOBuffer)
     return data
 end
 
+"Internal method. Copies the data from the IOBuffer to a string, leaving the
+IOBuffer in an unusable (erroneous) state.
```

Comment:
How so? It looks like this doesn't modify the iobuffer at all?

Reply:
Fixed in the latest commit - it still no longer modifies the buffer, but it now creates a string without copying (and without resetting the buffer to a usable state), which means mutating the buffer is illegal - hence the `unsafe_` name.